tasks/broad/ImputationTasks.wdl

```diff
@@ -0,0 +1,741 @@
+version 1.0
+
+task CalculateChromsomeLength {
+	input {
+		File ref_dict
+		Int chrom
+		String gatk_docker
+	}
+	command {
+		grep -P "SN:~{chrom}\t" ~{ref_dict} | sed 's/.*LN://' | sed 's/\t.*//'
+	}
+	runtime {
+		docker: gatk_docker
+		disks: "local-disk 100 HDD"
+		memory: "2 GB"
+	}
+	output {
+		Int chrom_length = read_int(stdout())
+	}
+}
+
+task GenerateChunk {
+	input {
+		Int start
+		Int end
+		String chrom
+		String basename
+		String vcf
+		String vcf_index
+		Int disk_size = 400 # not sure how big the disk size needs to be since we aren't downloading the entire VCF here
+		String gatk_docker
+	}
+	command {
+		gatk SelectVariants -V ~{vcf} --select-type-to-include SNP --max-nocall-fraction 0.1 -xl-select-type SYMBOLIC \
+		--select-type-to-exclude MIXED --restrict-alleles-to BIALLELIC -L ~{chrom}:~{start}-~{end} -O ~{basename}.vcf.gz --exclude-filtered true
+	}
+	runtime {
+		docker: gatk_docker
+		disks: "local-disk " + disk_size + " HDD"
+		memory: "8 GB"
+	}
+	parameter_meta {
+		vcf: {
+			description: "vcf",
+			localization_optional: true
+		}
+		vcf_index: {
+			description: "vcf index",
+			localization_optional: true
+		}
+	}
+	output {
+		File output_vcf = "~{basename}.vcf.gz"
+		File output_vcf_index = "~{basename}.vcf.gz.tbi"
+	}
+}
+
+task CountChunks {
+	input {
+		File vcf
+		File vcf_index
+		File panel_vcf
+		File panel_vcf_index
+		Int disk_size =ceil(2*size([vcf, vcf_index, panel_vcf, panel_vcf_index], "GB"))
+		String gatk_docker
+	}
+	command <<<
+		echo $(gatk CountVariants -V ~{vcf}  | sed 's/Tool returned://') > var_in_original
+		echo $(gatk  CountVariants -V ~{vcf} -L ~{panel_vcf}  | sed 's/Tool returned://') > var_in_reference
+		echo ${var_in_reference} " * 2 - " ${var_in_original} "should be greater than 0 AND " ${var_in_reference} "should be greater than 3"
```

Comment:
Is this just informational/debugging code?

Reply:
the final printout is, yes.